interview-questions.md

```diff
@@ -8,62 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: An argument is something that needed in order to log a return to a specific set of information, while a parameters is something that is placed within a function and used as input for our argument
 
+Researched answer: An argument is the actual value, while as a parameter is an input placed within a function
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: .map() takes in the predefined parameters from the original variable, the original parameters are required and the new variable is not
 
+Researched answer: Value, Index, and Array. Value is required and index, and array is optional
 
 3. What is the difference between map and filter?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: map pulls in predefined parameters, while filter allows you to specify what information should be returned
 
+Researched answer: map can navigate through and array and return one that is the same as the original, while a filter only returns what from the array what is in the specific condition
 
 4. What is the difference between a function and a method?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: A function is code that can work individually whereas a method can be contained in a class and can be linked
 
+Researched answer: Functions are a tool that can take in data and use it as inputs. A method operates the data within a class
 
 5. What is object destructuring?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Object destructuring is the process of breaking down an object in to a simple form
 
+Researched answer: Destructuring is a way of separating an object
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
-
-  Researched answer:
-
+Your answer: Being that hoisting moving moving up am going to guess that it is the process of moving code up
 
+Researched answer: The process of moving functions, variables, and classes to the top
```

Comment:
Correct, hoisting is javascript's way of trying to move variables and functions to the top so that it can run if they are defined later in the code.